gcloud/monitoring/group.py

```diff
@@ -0,0 +1,502 @@
+
+        The groups are returned in order, starting with the immediate parent
+        and ending with the most distant ancestor. If the specified group has
+        no immediate parent, the results are empty.
+
+        :rtype: list of :class:`~gcloud.monitoring.group.Group`
+        :returns: A list of group instances.
+        """
+        return self._list(self.client, ancestors_of_group=self.name)
+
+    def descendants(self):
+        """Lists all descendants of this group via a ``GET`` request.
+
+        This returns a superset of the results returned by the :meth:`children`
+        method, and includes children-of-children, and so forth.
+
+        :rtype: list of :class:`~gcloud.monitoring.group.Group`
+        :returns: A list of group instances.
+        """
+        return self._list(self.client, descendants_of_group=self.name)
+
+    def members(self, filter_string=None, end_time=None, start_time=None):
+        """Lists all members of this group via a ``GET`` request.
+
+        If no end_time and start_time is provided then the group membership
+        over the last minute is returned.
+
+        Examples::
+
+            To get current members that are Compute Engine VM instances:
+
+                group.members('resource.type = "gce_instance"')
+
+            To get historical members that existed between 4 and 5 hours ago:
+
+                import datetime
+                t1 = datetime.datetime.utcnow() - datetime.timedelta(hours=4)
+                t0 = t1 - datetime.timedelta(hours=1)
+                group.members(end_time=t1, start_time=t0)
+
+
+        :type filter_string: string or None
+        :param filter_string:
+            An optional list filter describing the members to be returned. The
+            filter may reference the type, labels, and metadata of monitored
+            resources that comprise the group. See the `filter documentation`_.
+
+        :type end_time: :class:`datetime.datetime` or None
+        :param end_time:
+            The end time (inclusive) of the time interval for which results
+            should be returned, as a datetime object.
+
+        :type start_time: :class:`datetime.datetime` or None
+        :param start_time:
+            The start time (exclusive) of the time interval for which results
+            should be returned, as a datetime object.  If not specified, the
+            interval is a point in time.
+
+        :rtype: list of :class:`~gcloud.monitoring.resource.Resource`
+        :returns: A list of resource instances.
+
+        :raises:
+            :exc:`ValueError` if the ``start_time`` is specified, but the
+            ``end_time`` is missing.
+
+        .. _filter documentation:
+            https://cloud.google.com/monitoring/api/v3/filters#group-filter
+        """
+        if start_time is not None and end_time is None:
+            raise ValueError('If "start_time" is specified, "end_time" must '
+                             'also be specified')
```

Comment:
Shouldn't there be a corresponding `end_time is not None and start_time is None`?

Don't you really want to check `int(start_time is None) + int(end_time is None) == 1`?